### ISSUE TITLE
Put "fat" JAR on a diet?

### ISSUE THREAD

**Issue by deiruch** [epfl-lara/stainless#1020]
This is a low-priority feature request. GitHub limits individual file sizes by default to 100MB. Because we commit the fat JAR to our GitHub repo (yes, I know...), its size is limited to 100MB as well. Current master releases weigh in above 100MB, so I manually remove parts which are not strictly required for verification and/or GenC. Currently I remove the bundled Z3 solver and the ICU library, which saves ~20MB. Looking at the Fat JAR, it appears that there are many dependencies (PDF generators, Emoji lists and others). I don't know how these end up in the JAR - or whether they are required.

TL;DR - I'd appreciate a reduction of the fat JAR size, if there's a simple way to do so.

**Comment by deiruch** [comment]
Z3 is included twice in the Linux build, once as library, once as external solver.
